src/managed_data/client.py

```diff
@@ -123,13 +157,24 @@ def invoke(
     ) -> Union[dict, list]:
         """core call to API endpoint"""
 
+        url = urljoin(self.api_url, endpoint)
+
         response = requests.post(
-            urljoin(self.api_url, endpoint),
+            url,
             headers=self.headers,
             json=data,
         )
 
-        return _check_response(response)
+        try:
+            response = _check_response(response)
+        except Exception as e:
+            print(f"URL: {url}")
+            print(f"headers: {self.headers}")
+            print(f"data: {data}")
+
+            raise e
+
+        return response
 
 
 class MockClient(Client):
```

Comment:
Why put this in the source code? I usually put resources used for testing either in the tests themselves or in a module specifically with resources for testing.

Reply:
that is a very good point. it should be in tests. i'll send you a new PR with that fixed